stackslib/src/chainstate/stacks/boot/pox-4-vote.clar

```diff
@@ -0,0 +1,56 @@
+;;
+;; @contract voting for the aggregate public key
+;;
+
+;; maps dkg round and signer to proposed aggregate public key
+(define-map votes {reward-cycle: uint, round: uint, signer: principal} {aggregate-public-key: (buff 33), reward-slots: uint})
+;; maps dkg round and aggregate public key to weights of signers supporting this key so far
+(define-map tally {reward-cycle: uint, round: uint, aggregate-public-key: (buff 33)} uint)
+
+(define-constant err-not-allowed (err u10000))
+(define-constant err-incorrect-reward-cycle (err u10001))
+(define-constant err-incorrect-round (err u10002))
+(define-constant err-invalid-aggregate-public-key (err u10003))
+(define-constant err-duplicate-vote (err u10004))
+(define-constant err-invalid-burn-block-height (err u10005))
+
+(define-data-var last-round uint u0)
+(define-data-var is-state-1-active bool true)
+(define-data-var state-1 {reward-cycle: uint, round: uint, aggregate-public-key: (optional (buff 33)),
+    total-votes: uint}  {reward-cycle: u0, round: u0, aggregate-public-key: none, total-votes: u0})
+(define-data-var state-2 {reward-cycle: uint, round: uint, aggregate-public-key: (optional (buff 33)),
+    total-votes: uint}  {reward-cycle: u0, round: u0, aggregate-public-key: none, total-votes: u0})
+
+;; get voting info by burn block height
+(define-read-only (get-info (height uint))
+    (ok (at-block (unwrap! (get-block-info? id-header-hash height) err-invalid-burn-block-height) (get-current-info))))
+
+;; get current voting info
+(define-read-only (get-current-info)
+    (if (var-get is-state-1-active) (var-get state-1) (var-get state-2)))
+
+(define-read-only (get-signer-public-key (signer principal) (reward-cycle uint))
+    ;; TODO replace with contract-call to pox-4::get-signer-public-key
```

Comment:
This might actually be in a `.signers` contract, which the node will populate with the reward set information when it begins a new reward cycle.  Then, it'll be cheaper to do this contract call since the `.signers` code body will be substantially shorter.